src/components/__tests__/SavedTranslations.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { shallowMount, Wrapper } from "@vue/test-utils";
+import SavedTranslations from "@/components/SavedTranslations.vue";
+import { ITranslation } from "@/components/types";
+import { Languages } from "@/languages";
+import Button from "@/components/Button.vue";
+
+jest.spyOn(Object.getPrototypeOf(window.localStorage), "getItem");
+jest.spyOn(Object.getPrototypeOf(window.localStorage), "setItem");
+
+describe("SavedTranslations", () => {
+  const translation: ITranslation = {
+    id: 1,
+    inputText: "привіт",
+    translation: "hi",
+    targetLang: Languages.EN,
+    sourceLang: Languages.UK,
+  };
+  const secondTranslation: ITranslation = {
+    id: 2,
+    inputText: "жінка",
+    translation: "woman",
+    targetLang: Languages.EN,
+    sourceLang: Languages.UK,
+  };
+  let wrapper: Wrapper<SavedTranslations>;
+  let vm: any;
+
+  beforeEach(() => {
+    wrapper = shallowMount(SavedTranslations, {
+      propsData: {
+        openSidebar: true,
+        translation,
+      },
+      stubs: {
+        Button,
+      },
+    });
+    vm = wrapper.vm;
+  });
+
+  it("should exists the component", () => {
+    expect(wrapper.findComponent(SavedTranslations).exists()).toBeTruthy();
+  });
+
+  it("should render with props", () => {
+    expect(wrapper.props("openSidebar")).toBeTruthy();
+    expect(wrapper.props("translation")).toEqual(translation);
+  });
+
+  it("should set class to the component", async () => {
+    const asideEl = wrapper.find("aside");
+    expect(asideEl.classes()).toContain("translations");
+    await wrapper.setProps({
+      openSidebar: false,
+    });
+    expect(asideEl.classes()).toEqual(
+      expect.arrayContaining(["translations", "hide"])
+    );
+  });
+
+  it("should get/set translations from localStorage", () => {
+    localStorage.setItem("translations", JSON.stringify([translation]));
+    vm.getTranslations();
+    const translationItems = JSON.parse(
+      localStorage.getItem("translations") || "{}"
+    );
+    expect(vm.translations).toEqual(translationItems);
+  });
+
+  it("should emit click event when clicked", async () => {
+    const btn = wrapper.findComponent(Button);
+    await btn.trigger("click");
+    expect(wrapper.emitted().hide).toBeTruthy();
+    const resultHide = vm.hide();
+    expect(resultHide).toBe(false);
+  });
+
+  it("should to push to translations a new prop", async () => {
+    expect(vm.translations.length).toEqual(1);
```

Comment:
if I delete [this](https://github.com/pakoval/translator-app/pull/8/files#diff-6b740b0c3ae818957033ef341f4e9a2bcd8ff2b51f415b80463efa04d9abb804R61) test this one won't work
FYI